src/Controller/FullCalendarController.php

```diff
@@ -0,0 +1,413 @@
+    $date_obj = new DrupalDateTime($date, \Drupal::configFactory()->get('system.date')->get('timezone')['default']);
+    return substr($date_obj->format('c', ['timezone' => 'UTC']), 0, 19);
+  }
+
+  /**
+   * Retrieves a list of published branches from the 'node_field_data' table.
+   *
+   * This method executes a database query to select node IDs (nid) and titles
+   * from the 'node_field_data' table where nodes are of the 'branch' content
+   * type and have a status indicating they are published. The results are
+   * ordered by the title of the node to ensure a consistent and alphabetical
+   * ordering of branches.
+   *
+   * Additionally, the query is tagged with 'openy_home_branch_get_locations'
+   * and 'node_access' to ensure that any alterations or access checks specific
+   * to the site's configuration or modules are applied, respecting the Drupal
+   * node access control system.
+   *
+   * @return array
+   *   An associative array where the key is the node ID (nid) of the branch,
+   *   and the value is the title of the branch.
+   */
+  public function getBranches() {
+    $query = $this->database->select('node_field_data', 'n');
+    $query->fields('n', ['nid', 'title']);
+    $query->condition('n.status', NodeInterface::PUBLISHED);
+    $query->condition('n.type', 'branch');
+    $query->orderBy('n.title');
+    $query->addTag('openy_home_branch_get_locations');
+    $query->addTag('node_access');
+
+    return $query->execute()->fetchAllKeyed();
+  }
+
+  /**
+   * Provides a JSON response containing a list of branches.
+   *
+   * This method leverages the `getBranches` method to retrieve an associative
+   * array of branch IDs and titles, and then encapsulates this array within a
+   * JSON response.
+   *
+   * @return JsonResponse
+   *   A JsonResponse object that contains the branches in an associative array
+   *   format, where each branch ID is a key, and its corresponding title is
+   *   the value.
+   */
+  public function getBranchesOptions(): JsonResponse {
+    return new JsonResponse($this->getBranches());
+  }
+
+  /**
+   * Provides a JSON response containing a list of published classes.
+   *
+   * This method executes a database query to select node IDs (nid) and titles
+   * from the 'node_field_data' table where nodes are of the 'class' content
+   * type and have a status indicating they are published.
+   *
+   * @return JsonResponse
+   *   A JsonResponse object that contains the classes in an associative array
+   *   format, where each class ID (nid) is a key, and its corresponding title
+   *   is the value.
+   */
+  public function getClassesOptions(): JsonResponse {
+    $query = $this->database->select('node_field_data', 'n');
+    $query->fields('n', ['nid', 'title']);
+    $query->condition('n.status', NodeInterface::PUBLISHED);
+    $query->condition('n.type', 'class');
+    $query->orderBy('n.title');
+    $res = $query->execute()->fetchAllKeyed();
```

Comment:
What if we have 10000+ classes?
How this behaves on mobile?
Can it be paginated?

Reply:
It already works as select2, that is, both autocomplete and select.
What should be the limit?

Reply:
100 items?

Reply:
Limited with 1oo items for now.